src/main/scala/io/github/spark_redshift_community/spark/redshift/RedshiftCatalog.scala

```diff
@@ -45,24 +46,17 @@ class RedshiftCatalog extends JDBCTableCatalog {
         }
       }
   
-  private def getTableName(ident: Identifier): String = {
-    (ident.namespace() :+ ident.name()).map(dialect.quoteIdentifier).mkString(".")
-  }
   override def invalidateTable(ident: Identifier): Unit = {
-    // TODO  When refresh table, then drop the s3 folder
+    import org.apache.hadoop.fs.{FileSystem, Path}
+    import java.net.URI
+    val parameters = Parameters.mergeParameters(options.parameters.toMap)
+    val rootDir = parameters.rootTempDir
+    val tableName = (ident.namespace() :+ ident.name()).map(dialect.quoteIdentifier).mkString(".")
+    val tableHash = tableName.hashCode.toString
+    val tablePath = Utils.fixS3Url(Utils.joinUrls(rootDir, tableHash))
+    val fs = FileSystem.get(URI.create(tablePath), spark.sparkContext.hadoopConfiguration)
+    logWarning(s"Invalidating cache for $tableName by deleting $tablePath recursively." +
+               s" Caution: this might break concurrent queries using cache on that table")
+    fs.delete(new Path(tablePath), true)
   }
-  override def dropTable(ident: Identifier): Boolean =
-    throw new UnsupportedOperationException("Drop table is not supported.")
-  override def renameTable(oldIdent: Identifier, newIdent: Identifier): Unit =
-    throw new UnsupportedOperationException("Rename table is not supported.")
-  override def createTable(ident: Identifier, schema: StructType, partitions: Array[Transform],
-    properties: util.Map[String, String]): SparkTable = {
-    logWarning("useing create table catalog")
-    new RedshiftDataSourceV2().getTable(new CaseInsensitiveStringMap(properties), schema)
-  }
-  
-  override def alterTable(ident: Identifier, changes: TableChange*): SparkTable =
-    throw new UnsupportedOperationException("Purge table is not supported.")
-  override def dropNamespace(namespace: Array[String]): Boolean =
-    throw new UnsupportedOperationException("Drop namespace is not supported.")
```

Comment:
did you mean to drop these?  Raising the exception makes sense to me still.

Reply:
By droping those the jdbccatalog methods are used. By mean the redshift driver handle them with jdbc call. So yes dropping those methods here, allow usage of them paradoxally

Reply:
neat.  🪄